Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person/GetNearestDrivers.hs

```diff
@@ -110,3 +121,55 @@ getNearestDrivers cityServiceTiers serviceTiers fromLocLatLong radiusMeters merc
         mkDriverResult mbDefaultServiceTierForDriver person vehicle info dist cityServiceTiersHashMap serviceTier = do
           serviceTierInfo <- HashMap.lookup serviceTier cityServiceTiersHashMap
           Just $ NearestDriversResult (cast person.id) person.deviceToken person.language info.onRide (roundToIntegral dist) vehicle.variant serviceTier (maybe 0 (\d -> d.priority - serviceTierInfo.priority) mbDefaultServiceTierForDriver) serviceTierInfo.isAirConditioned location.lat location.lon info.mode person.clientSdkVersion person.clientBundleVersion person.clientConfigVersion person.clientDevice person.backendConfigVersion person.backendAppVersion
+
+scheduledRideFilter :: (MonadFlow m, MonadTime m, LT.HasLocationService m r, ServiceFlow m r) => DST.CurrentSearchInfo -> Id Merchant -> Id DMOC.MerchantOperatingCity -> Bool -> Bool -> DTC.TransporterConfig -> DriverInfo.DriverInformation -> m Bool
+scheduledRideFilter currentSearchInfo merchantId merchantOpCityId isRental isIntercity transporterConfig driverInfo = do
+  now <- getCurrentTime
+  let minimumScheduledBookingLeadTimeInSecs = intToNominalDiffTime (transporterConfig.minmRentalAndScheduledBookingLeadTimeHours * 3600)
+      scheduledRideFilterExclusionThresholdInSecs = intToNominalDiffTime (transporterConfig.scheduledRideFilterExclusionThresholdHours * 3600)
+      haveScheduled = isJust driverInfo.latestScheduledBooking
+  if
+      | haveScheduled && isIntercity -> return False
+      | haveScheduled && isRental -> return $ canTakeRental driverInfo.latestScheduledBooking now minimumScheduledBookingLeadTimeInSecs
+      | checkUsingTime driverInfo.latestScheduledBooking now scheduledRideFilterExclusionThresholdInSecs -> do
+        dropLoc <- currentSearchInfo.dropLocation & fromMaybeM (InternalError "Error: dropLocation not found in currentSearchInfo.")
+        scheduledPickup <- driverInfo.latestScheduledPickup & fromMaybeM (InternalError "Error: latestScheduledPickup not found in driverInfo.")
```

Comment:
This could be Nothing, right? Why to throw error instead of that we can just return True.

Reply:
rental case is handled above so i think  we always get drop location .

Reply:
`driverInfo.latestScheduledPickup`  I am referring to this.

Reply:
it contains the pickup location of the upcoming scheduled ride